app/controllers/application_controller/explorer.rb

```diff
@@ -2,6 +2,19 @@
 module ApplicationController::Explorer
   extend ActiveSupport::Concern
 
+  def build_replaced_trees(replace_trees, valid_values)
+    return [] unless replace_trees
+    valid_values.find_all { |tree| replace_trees.include?(tree) }
+                .collect { |tree| try_build_tree(tree) }
+                .compact
+  end
+
+  def try_build_tree(tree_symbol)
+    method_name = "build_#{tree_symbol}_tree"
+    return unless respond_to?(method_name, true)
```

Comment:
I'm a little bit afraid that someone might try to remove the `build_*_tree` methods as they might seem unused after a grepping. Maybe it's over-engineering, but I'd rather keep a list of these methods here and check against. What do you think?

Reply:
Well I prefer not. I'd like to unify it a bit further and then remove the build_*tree methods calling directly the tree builder.

There's no value and little information in the build_*tree methods

Reply:
Good enough for me :+1: